### DISCUSSION TITLE
Invalid id of type string for document type Xyz

### DISCUSSION THREAD

**Discussion by TatChu** [JasperFx/marten#3238] [Q&A]
I'm trying to use the revision approach for the concurrency problem.
This is my entity
```csharp
public class public class Notification : IRevisioned
{
    public string Id { get; set; }
   // other props
    public int Version { get; set; }
}
```
Then when updating the document, I got this exception
`Marten.Exceptions.DocumentIdTypeMismatchException: Invalid id of type string for document type Domain.Notification`.
Unsure what went wrong with the above code?

Version: 7.16.0

**Comment by jeremydmiller** [comment]
@TatChu Can you please use Discord for questions in the future?

Did you change this document at some point in time? Do you possibly have stale versions of pre-generated code around?
